### PR TITLE
fix(terraform): Fix CKV_AZURE_227 for Azure V4

### LINKED ISSUE #6881: CKV_AZURE_227 don't working with AzureRM v4 changes on AKS
**Describe the issue**
The current CKV_AZURE_227 checks for ‘enable_host_encryption’ in the resource ‘azurerm_kubernetes_cluster’ and ‘azurerm_kubernetes_cluster_node_pool’. With the update to Azurerm v4, the variable changes from ‘enable_host_encryption’ to ‘host_encryption_enabled’. With this variable the check CKV_AZURE_227 fails.

**Examples**
resource "azurerm_kubernetes_cluster" "this" {
  default_node_pool {
    name = "default"
    enable_host_encryption       = true
}

Check: CKV_AZURE_227: "Ensure that the AKS cluster encrypt temp disks, caches, and data flows between Compute and Storage resources"
	FAILED for resource: module.aks.azurerm_kubernetes_cluster.this
	File: /plan.json:2181-2466
	Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/azure-policies/azure-kubernetes-policies/bc-azure-227

**Version (please complete the following information):**
 - Checkov Version 3.2.320

**Additional context**
--

### LINKED ISSUE #6902: CKV_AZURE_145 doesn't acknowledge that TLS can be set to 1.3 and expects 1.2 as minimum
**Describe the issue**
Checkov fails incorrectly when the minimum_tls_version in azurerm_linux_function_app resource is set to 1.3, which is possible now in azurerm provider version 4.13.0

**Examples**
Terraform code example:

`site_config {
  minimum_tls_version = "1.3"
}`

**Version (please complete the following information):**
 - 3.2.334

**Additional context**

### PR DESCRIPTION
**Pull request by tsmithv11**
# User description
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    We use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the other types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Each prefix should be accompanied by a scope that specifies the targeted framework. If uncertain, use 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Fixes #6881 
Fixes #6902 

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes

---

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
<p>This pull request includes updates to security checks, refactoring of existing checks, and additional test cases for Azure resources. The changes aim to enhance security measures, improve check accuracy, and ensure comprehensive test coverage.</p>
    <table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/bridgecrewio/checkov/6906?tool=ast&topic=Check+Refactoring>Check Refactoring</a>
        </td><td>Refactors and expands existing checks for better accuracy and coverage<details><summary>Modified files (10)</summary><ul><li>tests/terraform/graph/checks/resources/VAisEnabledInStorageAccount/expected.yaml</li>
<li>tests/terraform/checks/resource/azure/test_SQLServerNoPublicAccess.py</li>
<li>tests/arm/checks/resource/test_MySQLPublicAccessDisabled.py</li>
<li>tests/terraform/graph/checks/resources/AzureSQLserverNotOverlyPermissive/expected.yaml</li>
<li>tests/terraform/graph/checks/resources/AzureSQLserverNotOverlyPermissive/main.tf</li>
<li>tests/terraform/checks/resource/azure/test_AKSNodePublicIpDisabled.py</li>
<li>tests/terraform/graph/checks/resources/VAisEnabledInStorageAccount/main.tf</li>
<li>checkov/terraform/checks/graph_checks/azure/AzureSQLserverNotOverlyPermissive.yaml</li>
<li>checkov/terraform/checks/graph_checks/azure/VAisEnabledInStorageAccount.yaml</li>
<li>tests/terraform/checks/resource/azure/test_AKSEncryptionAtHostEnabled.py</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>channy-katz</td><td>feat-arm-MySQLPublicAc...</td><td>May 22, 2024</td></tr>
<tr><td>tdefise</td><td>fix-terraform-Fixing-i...</td><td>October 16, 2023</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/bridgecrewio/checkov/6906?tool=ast&topic=Security+Enhancements>Security Enhancements</a>
        </td><td>Improves security measures for Azure resources including TLS versions, public access controls, and encryption<details><summary>Modified files (13)</summary><ul><li>checkov/terraform/checks/resource/azure/SQLServerNoPublicAccess.py</li>
<li>checkov/arm/checks/resource/MySQLPublicAccessDisabled.py</li>
<li>checkov/terraform/checks/resource/azure/AKSEncryptionAtHostEnabled.py</li>
<li>tests/arm/checks/resource/example_FunctionAppMinTLSVersion/pass4.json</li>
<li>checkov/arm/checks/resource/FunctionAppMinTLSVersion.py</li>
<li>checkov/terraform/checks/resource/azure/AKSNodePublicIpDisabled.py</li>
<li>tests/arm/checks/resource/example_TestMySQLPublicAccessDisabled/pass2.json</li>
<li>tests/terraform/checks/resource/azure/example_SQLServerNoPublicAccess/main.tf</li>
<li>checkov/terraform/checks/resource/azure/FunctionAppMinTLSVersion.py</li>
<li>tests/arm/checks/resource/example_TestMySQLPublicAccessDisabled/fail2.json</li>
<li>tests/terraform/checks/resource/azure/example_AKSNodePublicIpDisabled/main.tf</li>
<li>tests/terraform/checks/resource/azure/example_AKSEncryptionAtHostEnabled/main.tf</li>
<li>tests/terraform/checks/resource/azure/example_FunctionAppMinTLSVersion/main.tf</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>channy-katz</td><td>feat-arm-MySQLPublicAc...</td><td>May 22, 2024</td></tr>
<tr><td>rutiNalenger</td><td>feat-arm-add-CKV_AZURE...</td><td>May 22, 2024</td></tr></table></details></td></tr></table>This pull request is reviewed by Baz. Join @tsmithv11 and the rest of your team on <a href=https://baz.co/login>(Baz)</a>.

### CODE REVIEW

**Review by SteveVaknin — APPROVED** [review]

File: checkov/terraform/checks/resource/azure/FunctionAppMinTLSVersion.py [hunk 1]
```diff
@@ -20,17 +22,17 @@ def __init__(self):
...
 
-    def get_inspected_key(self):
+    def get_inspected_key(self) -> str:
         if self.entity_type in ("azurerm_function_app", "azurerm_function_app_slot"):
             return "site_config/[0]/min_tls_version"
         else:
             return "site_config/[0]/minimum_tls_version"
 
-    def get_expected_value(self):
+    def get_expected_value(self) -> Any:
```
Comment (talazuri) [inline checkov/terraform/checks/resource/azure/FunctionAppMinTLSVersion.py:31]: Why Any?

Comment (tsmithv11) [reply checkov/terraform/checks/resource/azure/FunctionAppMinTLSVersion.py]: Changed it to `float`

File: checkov/terraform/checks/resource/azure/AKSEncryptionAtHostEnabled.py [hunk 2]
```diff
@@ -22,14 +22,23 @@ def __init__(self) -> None:
...
-    def get_inspected_key(self) -> str:
+    def scan_resource_conf(self, conf):
         if self.entity_type == "azurerm_kubernetes_cluster":
-            return "default_node_pool/[0]/enable_host_encryption"
+            if conf.get('default_node_pool'):
+                node_pool = conf['default_node_pool'][0]
+                if (node_pool.get('enable_host_encryption') == [True] or
+                        node_pool.get('host_encryption_enabled') == [True]):
+                    return CheckResult.PASSED
+            self.evaluated_keys = ['default_node_pool/[0]/enable_host_encryption',
```
Comment (talazuri) [inline checkov/terraform/checks/resource/azure/AKSEncryptionAtHostEnabled.py:34]: Do we need to update the evaluated_keys here? What is the reason?

**Review by talazuri — APPROVED** [review]
LGTM, some minor comments

File: checkov/terraform/checks/resource/azure/AKSEncryptionAtHostEnabled.py (hunk 2 above)
Comment (tsmithv11) [reply checkov/terraform/checks/resource/azure/AKSEncryptionAtHostEnabled.py:34]: For highlighting in the platform (and fix suggestions, but not relevant for this check)

**Pull request merged**